src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -291,7 +297,8 @@ protected internal void RuntimeLog(byte[] state)
         /// <param name="state">The arguments of the event.</param>
         protected internal void RuntimeNotify(byte[] eventName, Array state)
         {
-            if (eventName.Length > MaxEventName) throw new ArgumentException(null, nameof(eventName));
+            if (eventName.Length > MaxEventName)
```

Comment:
I don't think ANY exception in methods that modifies contract state (including notifications) should be catchable